server/ctools_direct_response.rb

```diff
@@ -0,0 +1,67 @@
+###### CTools direct URL provider ################
+# Format the CTools direct response into json format that UI expects.
+
+require_relative './Logging'
+require 'rest-client'
+
+# Handle interpretation and reformatting of a CTools direct call.
+# The incoming data will NOT have the WAPI wrapper.
+
+## Output format is defined in: https://docs.google.com/document/d/1KfhK6aMW1FdZdQwzTj5UO_zHq1qG6Fnpq80yWFV8hN4/edit
+
+class CToolsDirectResponse
+
+  # Store the body of the response.  All output from CTools direct goes through this class.
+  attr_accessor :body_string, :body_json
+
+  ## TODO: allow input as string or json
+
+  # Input is in string format
+  def initialize(body)
+    @body_string = body
+    @body_json = JSON.parse(body)
+
+    #dump
+  end
+
+  # return the body formatted as todolms json
+  def toDoLms
+    collection = @body_json['dash_collection']
+    return Array.new() if collection.nil?
+    # process all the assignments.
+    collection.map { |dash_entry| extractAssignment(dash_entry) }
+  end
+
+  def extractAssignment(assignment)
+
+    # create a single assignment object in known format for UI.
+    assign = Hash.new()
+
+    assign[:title] = assignment['calendarItem']['title']
+    assign[:grade] = ''
```

Comment:
if the "grade" and "done" always have empty String value, why do we need those two attributes?